packages/medusa/src/services/price-list.ts

```diff
@@ -412,6 +412,13 @@ class PriceListService extends TransactionBaseService {
         })
       )
 
+      if (priceListId) {
+        const variantsWithPriceList = variants.filter((variant) =>
+          variant.prices.some((price) => price.price_list_id === priceListId)
+        )
+        return [variantsWithPriceList, variantsWithPriceList.length]
```

Comment:
I suggest you to instead look into the repository to avoid having to do that. It seams that the repository takes into account the price list id

Reply:
I assume the expected behavior of this method is to give variants that are in the price list.
(As it is not mentioned in the documentation that giving the price list id will either filter or add the prices to variants)
So,
This is my understanding of this code. Please correct me if I'm wrong.

1. At line 396 all variants are queried and it takes the selector `id` (variant_id) to filter further. Since it doesn't return prices we cannot know whether the variant is included in a price list or not
2. At line 403 we query prices for all the variants within the `priceListId` and append the prices to respective variants. 
4. Now here all the variants (in the database) are returned which we queried in the first step along with some of them having prices (with `priceListId`).


![image](https://user-images.githubusercontent.com/47112778/231411217-f4c46cf5-98cb-4a06-aa6b-94f6dff5ffb0.png)

Reply:
oh okay, I should close this PR then?

ps.
Found Medusa from Explore tab, a really great tool for commerce. Kudos!
If is there any opening for Software Engineer with 3yr+ experience, I would definitely be a part of this mission.